dotCMS/src/main/java/com/dotcms/rendering/velocity/services/DotResourceLoader.java

```diff
@@ -0,0 +1,158 @@
+package com.dotcms.rendering.velocity.services;
+
+
+import com.dotcms.repackage.org.apache.commons.collections.ExtendedProperties;
+
+import com.dotmarketing.business.APILocator;
+import com.dotmarketing.business.CacheLocator;
+import com.dotmarketing.util.Logger;
+import com.dotmarketing.util.PageMode;
+import com.dotmarketing.util.UtilMethods;
+
+import java.io.InputStream;
+
+import org.apache.velocity.exception.ResourceNotFoundException;
+import org.apache.velocity.runtime.resource.Resource;
+import org.apache.velocity.runtime.resource.loader.ResourceLoader;
+
+public class DotResourceLoader extends ResourceLoader {
+
+
+    private static DotResourceLoader instance;
+
+
+    public DotResourceLoader() {
+        super();
+    }
+
+    
+    @Override
+    public InputStream getResourceStream(final String filePath) throws ResourceNotFoundException {
+        if (!UtilMethods.isSet(filePath)) {
+            throw new ResourceNotFoundException("cannot find resource");
+        }
+
+
+        final VelocityType type = VelocityType.resolveVelocityType(filePath);
+        synchronized (filePath.intern()) {
+
+
+
+            String paths = (filePath.charAt(0) == '/') ? filePath.substring(1, filePath.length()) : filePath;
+
+            final String[] path = paths.split("[/\\.]", -1);
+            final PageMode mode = PageMode.get(path[0]);
+            final String id1 = path[1].indexOf("_") > -1 ? path[1].substring(0, path[1].indexOf("_")) : path[1];
+            final String language = path[1].indexOf("_") > -1 ? path[1].substring(path[1].indexOf("_") + 1, path[1].length())
+                    : String.valueOf(APILocator.getLanguageAPI()
+                        .getDefaultLanguage()
+                        .getId());
+
+            final String id2 = path.length > 3 ? path[2] : null;
+
+
+            Logger.debug(this, "DotResourceLoader:\tInode: " + id1);
+
+            try {
+                switch (type) {
+                    case NOT_VELOCITY: {
+                        CacheLocator.getVeloctyResourceCache()
+                        .addMiss(path);
+                    throw new ResourceNotFoundException("Cannot find velocity file : " + path);
+                    }
+                    case CONTAINER: {
+                        return new ContainerLoader().writeObject(id1, id2, mode, language, filePath);
+                    }
+                    case TEMPLATE: {
+
+                        return new TemplateLoader().writeObject(id1, id2, mode, language, filePath);
+
+                    }
+                    case CONTENT: {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Extract this nested code block into a method. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1199)